src/pgduckdb.cpp

```diff
@@ -79,9 +94,15 @@ DefineCustomVariable(const char *name, const char *short_desc, char **var, GucCo
 }
 
 template <typename T>
-static void
+using GucTypeCheckHook = bool (*)(T *, void **, GucSource);
+
+template <typename T>
+using GucTypeAssignHook = void (*)(T, void *);
+
+template <typename T>
+void
```

Comment:
Let's put static back so it doesn't conflict with symbols from possibly other libraries.

```suggestion
static void
```

Reply:
So these are templates, so it won't conflict with anything, and I've put all the concrete implementation in the anonymous ns: https://github.com/duckdb/pg_duckdb/blob/f1fc8f4747e37daf0f9a23f26a50e46b287726fa/src/pgduckdb.cpp#L66